app/models/presentation.rb

```diff
@@ -1,4 +1,17 @@
 class Presentation < ApplicationRecord
-  has_many :person_presentations
-  has_many :people, through: :person_presentations
+  belongs_to :person, optional: true
+  belongs_to :event
+
+  validates :topic,     presence: true
+  validates :duration,  presence: true
+  validates :presenter, presence: true, unless: :person
+  validates :person,    presence: true, unless: :presenter
+
+  def presenter_name
```

Comment:
right! I originally had it part of the view, then moved it to the controller, and settled on the model. Forgot about writing a test for it. Thanks!